include/dali/core/mm/malloc_resource.h

```diff
@@ -130,16 +130,63 @@ class pinned_malloc_memory_resource : public pinned_async_resource {
   }
 
   bool do_is_equal(const memory_resource<memory_kind> &other) const noexcept override {
-    return dynamic_cast<const cuda_malloc_memory_resource*>(&other) != nullptr;
+    return dynamic_cast<const pinned_malloc_memory_resource*>(&other) != nullptr;
   }
 
  public:
-  static cuda_malloc_memory_resource &instance() {
-    static cuda_malloc_memory_resource inst;
+  static pinned_malloc_memory_resource &instance() {
```

Comment:
Fix for copy/paste?
Can we test this anyhow?

Reply:
Now we do :)